From 986100ba5f1a3f42b4ca573ae5356e79c5739060 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Wed, 4 Sep 2013 11:00:04 -0400
Subject: [PATCH 0139/1074] ENGR00278064 arm: dts: Add gpio keyboard support
 for i.mx6qdl sabreauto board

Add gpio keyboard support for i.mx6qdl sabreauto board,
they are home, back, program, vol+ and vol-.

Signed-off-by: Anson Huang <b20788@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 arch/arm/boot/dts/imx6qdl-sabreauto.dtsi |   43 ++++++++++++++++++++++++++++++
 1 files changed, 43 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
index c802454..39d1a32 100644
--- a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
@@ -22,6 +22,44 @@
 		reg = <0x10000000 0x80000000>;
 	};
 
+	gpio-keys {
+		compatible = "gpio-keys";
+		home {
+			label = "Home";
+			gpios = <&gpio1 11 0>;
+			linux,code = <102>; /* KEY_HOME */
+			gpio-key,wakeup;
+		};
+
+		back {
+			label = "Back";
+			gpios = <&gpio1 12 0>;
+			linux,code = <158>; /* KEY_BACK */
+			gpio-key,wakeup;
+		};
+
+		program {
+			label = "Program";
+			gpios = <&gpio2 12 0>;
+			linux,code = <362>; /* KEY_PROGRAM */
+			gpio-key,wakeup;
+		};
+
+		volume-up {
+			label = "Volume Up";
+			gpios = <&gpio2 15 0>;
+			linux,code = <115>; /* KEY_VOLUMEUP */
+			gpio-key,wakeup;
+		};
+
+		volume-down {
+			label = "Volume Down";
+			gpios = <&gpio5 14 0>;
+			linux,code = <114>; /* KEY_VOLUMEDOWN */
+			gpio-key,wakeup;
+		};
+	};
+
 	max7310_reset: max7310-reset {
 		compatible = "gpio-reset";
 		reset-gpios = <&gpio1 15 GPIO_ACTIVE_LOW>;
@@ -222,6 +260,11 @@
 				MX6QDL_PAD_EIM_EB0__GPIO2_IO28   0x80000000
 				MX6QDL_PAD_EIM_EB1__GPIO2_IO29	0x80000000
 				MX6QDL_PAD_DISP0_DAT23__GPIO5_IO17 0x80000000
+				MX6QDL_PAD_SD2_CMD__GPIO1_IO11 0x80000000
+				MX6QDL_PAD_SD2_DAT3__GPIO1_IO12 0x80000000
+				MX6QDL_PAD_SD4_DAT4__GPIO2_IO12 0x80000000
+				MX6QDL_PAD_SD4_DAT7__GPIO2_IO15 0x80000000
+				MX6QDL_PAD_DISP0_DAT20__GPIO5_IO14 0x80000000
 			>;
 		};
 	};
-- 
1.7.5.4

